src/sdk/namespace_merge.js

```diff
@@ -25,9 +25,18 @@ class NamespaceMerge {
     // OBJECT LIST //
     /////////////////
 
-    list_objects(params, object_sdk) {
-        return this._ns_map(ns => ns.list_objects(params, object_sdk))
-            .then(res => this._handle_list(res, params));
+    async list_objects(params, object_sdk) {
```

Comment:
It is great that we attempt to fix list_objects but I understand that we have this problem in all of our lists and even other operations (why not fix them too?). 
This is why I've coded an exception in Namespace_Merge that will do what you are trying to do.
The thing is that I've coded this for RPC responses but you can use it for SDK errors as well.
Take a look at `const reply = await this._ns_map(ns => ns.delete_object(params, object_sdk), EXCEPT_REASONS);`
Notice that we have `EXCEPT_REASONS` and take a look at the logic behind it.
You can use the same thing in your change without coding a separate method with error handling.

Reply:
Using _throw_if_any_failed_or_get_succeeded is not relevant in this case because of 2 cases:
1. Base case: The namespace bucket is using only one namespace resource and this only one is faulty (err= 'NoSuchBucket).
2. General case: All namespace resources of the namespace bucket are faulty (err= 'NoSuchBucket').
If i'll call your function, i'll get an empty array but for these cases I would like to throw the error 'NoSuchBucket'.

Reply:
tl;dr:
This suggestion won't work because I would like to throw an error if **all** the iterations fails on "NoSuchBucket " and the suggested function omits the errors information and does not answer this need.

Reply:
```    _throw_if_any_failed_or_get_succeeded(reply_array, except_reasons) {
        const failed = this._get_failed_responses(reply_array, except_reasons);
        if (!_.isEmpty(failed)) throw _.first(failed);
        const succeeded = this._get_succeeded_responses(reply_array);
        // Since we did not have any success and all of the errors were except_reasons we rely on the first error.
        if (_.isEmpty(succeeded)) throw _.first(reply_array).error;
        return succeeded;
    }
```
1) Base case: The only one is faulty err= 'NoSuchBucket', failed is empty because you did except that error,  succeeded is empty because the only one failed, then you return throw _.first(reply_array).error which is the err= 'NoSuchBucket' - works.
2) General case: Same as first one, all are faulty and you ignore because you did except that error, succeeded is empty because all failed, so you return the first error which is err= 'NoSuchBucket'.
3) Another case: Several are faulty with NoSuchBucket and one has an actual problem, you will return this one bucket.
4) Several are faulty and one is succeeded: you will return the succeeded if all faulty are except reason.

What doesn't work? I'm trying to figure out what I'm missing.

Reply:
**1-3: not a bug but it's not informative enough:**
When calling this function the returned error is not an S3 error and that leads for throwing InternalError instead of the specific S3Error with the specific message. 
4: **bug:** won't work if the succeeded one is empty, it will return the error and not an empty array.